### PR TITLE
Post snapshot directly to Percy agent from outside of the browser

### PR DESCRIPTION
**Pull request by anaulin**
In this implementation, the agent port and paths are hardcoded in the SDK. We should figure out a good way to expose those constants from `@percy/agent` for reuse in SDKs.

### CODE REVIEW

**Comment by djones** [comment]
I believe this is going to resolve the initial issue outlined in #24

File: lib/index.ts [hunk 1]
```diff
@@ -27,8 +28,45 @@ export async function percySnapshot(page: Page, name: string, options: any = {})
...
+    return true
+  }).catch((error) => {
+    return false
+  })
+}
+
+async function postDomSnapshot(name: string, domSnapshot: any, url: string, options: any) {
+  await Axios({
+    method: 'post',
+    url: 'http://localhost:5338/percy/snapshot',
```
[the remark this answers is not in the document]
Comment (djones) [reply lib/index.ts]: > I think a better solution would be exposing that information from @percy/agent in some cleaner way.

I like the sound of that. Wanna see what might be possible?

**Comment by anaulin** [comment]
@djones I've updated this PR with what the SDK-side would look like if we adopted a change like the one proposed in https://github.com/percy/percy-agent/pull/56

(Tests on this PR currently won't pass because it depends on those agent changes.)

File: lib/index.ts [hunk 2]
```diff
@@ -27,8 +27,29 @@ export async function percySnapshot(page: Page, name: string, options: any = {})
...
+  if (! await isAgentRunning()) {
+    return
+  }
+
+  const domSnapshot = await page.evaluate(function(name: string, options: any, clientInfo: string) {
+    const percyAgentClient = new PercyAgent({ clientInfo, handleAgentCommunication: false })
+    return percyAgentClient.snapshot(name, options)
   }, name, options, clientInfo())
+
+  const url = await page.evaluate(() => { return document.URL })
```
Comment (djones) [inline lib/index.ts]: We can use `page.url()` here

File: lib/index.ts [hunk 3]
```diff
@@ -27,8 +27,29 @@ export async function percySnapshot(page: Page, name: string, options: any = {})
...
   })
-  await page.evaluate(function(name: string, options: any, clientInfo: string) {
-    const percyAgentClient = new PercyAgent({ clientInfo })
-    percyAgentClient.snapshot(name, options)
+
+  if (! await isAgentRunning()) {
+    return
+  }
+
+  const domSnapshot = await page.evaluate(function(name: string, options: any, clientInfo: string) {
```
Comment (djones) [inline lib/index.ts]: as discussed, we'll remove the `clientInfo` references here since they don't actually take effect.

**Comment by toolmantim** [comment]
Woohoo! Thank you so much for the change, this is going to make using Percy SO much easier against our prod Docker images 😊❤️🙌🏼

**Pull request merged**